From 0cc8ed5abc41e0dd269e749794548645db0a4e97 Mon Sep 17 00:00:00 2001
From: Mashopy <eliasgheeraert@gmail.com>
Date: Sat, 23 Aug 2025 07:15:29 +0200
Subject: [PATCH 2/2] fastbootd: Always return false for GetDeviceLockStatus()

fenrir lks can still flash stuff through bootloader and fastboot

Change-Id: Ica59dab533555af1d12c62b9cc71de55d5b51146

[Shirayuki39]: Make it configurable via soong config

Signed-off-by: Shirayuki39 <lorddemecrius83@proton.me>
---
 fastboot/Android.bp         | 5 +++++
 fastboot/device/utility.cpp | 4 ++++
 2 files changed, 9 insertions(+)

diff --git a/fastboot/Android.bp b/fastboot/Android.bp
index 9e6d4132c..53bbcd8de 100644
--- a/fastboot/Android.bp
+++ b/fastboot/Android.bp
@@ -153,6 +153,11 @@ cc_binary {
         default: [],
     }),
 
+    cppflags: select(soong_config_variable("fastbootd", "bypass_lock_state"), {
+        true: ["-DFASTBOOT_BYPASS_LOCK_STATE"],
+        default: [],
+    }),
+
     srcs: [
         "device/commands.cpp",
         "device/fastboot_device.cpp",
diff --git a/fastboot/device/utility.cpp b/fastboot/device/utility.cpp
index d374305e3..3c5726d5b 100644
--- a/fastboot/device/utility.cpp
+++ b/fastboot/device/utility.cpp
@@ -195,7 +195,11 @@ std::vector<std::string> ListPartitions(FastbootDevice* device) {
 }
 
 bool GetDeviceLockStatus() {
+#ifdef FASTBOOT_BYPASS_LOCK_STATE
+    return false;
+#else
     return android::base::GetProperty("ro.boot.verifiedbootstate", "") == "green";
+#endif
 }
 
 bool UpdateAllPartitionMetadata(FastbootDevice* device, const std::string& super_name,
-- 
2.53.0

